scripts/devaccess/helper_scripts/3scale-dump.sh

```diff
@@ -0,0 +1,1314 @@
+        ${COMMAND} | grep -i "${THREESCALE_PROJECT}" 2> ${DUMP_DIR}/temp-cmd.txt | awk '{print $1}' > ${DUMP_DIR}/temp.txt
+
+        detect_error
+
+    else
+        ${COMMAND} 2> ${DUMP_DIR}/temp-cmd.txt | awk '{print $1}' | tail -n +2 > ${DUMP_DIR}/temp.txt
+
+        detect_error
+    fi
+}
+
+detect_error() {
+    if [[ -f ${DUMP_DIR}/temp-cmd.txt ]]; then
+        TEMP_CMD=$(< ${DUMP_DIR}/temp-cmd.txt)
+
+        if [[ -n ${TEMP_CMD} ]]; then
+            echo -e "Error on the Step ${STEP} (${STEP_DESC}):\n" >> ${DUMP_DIR}/errors.txt
+            cat ${DUMP_DIR}/temp-cmd.txt >> ${DUMP_DIR}/errors.txt
+            echo -e "\n" >> ${DUMP_DIR}/errors.txt
+        fi
+
+        /bin/rm -f ${DUMP_DIR}/temp-cmd.txt
+
+        unset TEMP_CMD
+    fi
+}
+
+read_obj() {
+    if [[ -z ${NOYAML} ]]; then
+        YAML="-o yaml"
+        INLINE_FILTER="| ${REDACT_SCRIPT}"
+    else
+        unset YAML
+        unset INLINE_FILTER
+    fi
+
+    while read OBJ; do
+        if [[ ! ${VALIDATE_PODS} == 1 ]]; then
+            FOUND=1
+
+        else
+            FOUND=0
+
+            for POD in "${THREEESCALE_PODS[@]}"; do
+                if ( [[ ${SUBSTRING} == 1 ]] && [[ "${OBJ}" == *"${POD}"* ]] ) || ( [[ "${OBJ}" == "${POD}" ]] ); then
+                    FOUND=1
+                fi
+            done
+        fi
+
+        if [[ ! ${FOUND} == 1 ]]; then
+            echo -e "\tSkipping: ${OBJ}"
+
+        else
+            if [[ ${VERBOSE} == 1 ]]; then
+                echo -e "\n\tProcess: ${OBJ}"
+            fi
+
+            if [[ ${COMPRESS} == 1 ]]; then
+                CONTAINERS=$(${COMMAND} ${OBJ} --template='{{range .spec.containers}}{{.name}}
+{{end}}')
+
+                for CONTAINER in ${CONTAINERS}; do
+                    if [[ ${PREVIOUS} == 1 ]]; then
+
+                        oc logs -p ${OBJ} --container=${CONTAINER} --timestamps > ${DUMP_DIR}/${NEWDIR}/previous-temp.txt 2>&1
+
+                        VALIDATE=$(< ${DUMP_DIR}/${NEWDIR}/previous-temp.txt head -n 1)
+
+                        if [[ ! "${VALIDATE,,}" == *"previous terminated container"* ]] && [[ ! "${VALIDATE,,}" == *"not found"* ]]; then
+                            cat ${DUMP_DIR}/${NEWDIR}/previous-temp.txt | ${COMPRESS_UTIL} -f - > ${DUMP_DIR}/${NEWDIR}/${OBJ}-${CONTAINER}.${COMPRESS_FORMAT}
+
+                        else
+                            echo -e "\n${VALIDATE}" >> ${DUMP_DIR}/${NEWDIR}/ignored-temp.txt
+                        fi
+
+                    else
+                        oc logs ${OBJ} --container=${CONTAINER} --timestamps 2>&1 | ${COMPRESS_UTIL} -f - > ${DUMP_DIR}/${NEWDIR}/${OBJ}-${CONTAINER}.${COMPRESS_FORMAT}
+                    fi
+                done
+
+                sleep 1.0
+
+            else
+                ${COMMAND} ${OBJ} ${YAML} ${INLINE_FILTER} >> ${DUMP_DIR}/${SINGLE_FILE} 2>&1
+                ${COMMAND} ${OBJ} ${YAML} ${INLINE_FILTER} > ${DUMP_DIR}/${NEWDIR}/${OBJ}.yaml 2>&1
```

Comment:
Without "eval" at the beginning of these commands, the pipe in $INLINE_FILTER is not being treated as an actual pipe and is instead being treated as an argument to $COMMAND (oc), resulting YAMLs ended up with an oc error message.

```suggestion
                eval ${COMMAND} ${OBJ} ${YAML} ${INLINE_FILTER} >> ${DUMP_DIR}/${SINGLE_FILE} 2>&1
                eval ${COMMAND} ${OBJ} ${YAML} ${INLINE_FILTER} > ${DUMP_DIR}/${NEWDIR}/${OBJ}.yaml 2>&1
```